activerecord/test/cases/relation/with_test.rb

```diff
@@ -57,9 +58,28 @@ def test_with_when_called_from_active_record_scope
       end
 
       def test_with_when_invalid_params_are_passed
-        assert_raise(ArgumentError) { Post.with }
         assert_raise(ArgumentError) { Post.with(posts_with_tags: nil).load }
-        assert_raise(ArgumentError) { Post.with(posts_with_tags: [Post.where("tags_count > 0")]).load }
+        assert_raise(ArgumentError) { Post.with(posts_with_tags: [Post.where("tags_count > 0"), 5]).load }
+      end
+
+      def test_with_when_passing_arrays
+        relation = Post
+          .with(posts_with_tags_or_comments: [
+            Post.where("tags_count > 0"),
+            Post.where("legacy_comments_count > 0")
+          ])
+          .from("posts_with_tags_or_comments AS posts")
+
+        assert_equal POSTS_WITH_TAGS_OR_COMMENTS, relation.order(:id).pluck(:id)
+      end
+
+      def test_with_recursive
+        # TODO: actually test recursive behavior
+        relation = Post
+          .with.recursive(posts_with_comments: Post.where("legacy_comments_count > 0"))
+          .from("posts_with_comments AS posts")
+
+        assert_equal POSTS_WITH_COMMENTS, relation.order(:id).pluck(:id)
```

Comment:
I'm not familiar with the test fixtures and could not find ones that lend themselves to testing actual recursive CTEs.